integration_tests/k8s/k8sapi.go

```diff
@@ -0,0 +1,226 @@
+package k8s
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"time"
+)
+
+// The wait time for all k8s API related operations
+const k8sWaitTime = time.Minute
+
+// Get all the pods in the given namespace along with their status
+func GetPods(clientSet *kubernetes.Clientset, ns string) (map[string]string, error) {
+	pods := make(map[string]string)
+	podList, err := clientSet.CoreV1().Pods(ns).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range podList.Items {
+		pods[pod.Name] = string(pod.Status.Phase)
+	}
+	return pods, nil
+}
+
+// Get all the nodes in the cluster along with their status
+func GetNodes(clientSet *kubernetes.Clientset) (map[string]string, error) {
+	nodes := make(map[string]string)
+	nodeList, err := clientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range nodeList.Items {
+		nodes[node.Name] = string(node.Status.Phase)
+	}
+	return nodes, nil
+}
+
+// Create all the objects specified in the NDM operator YAML
+// ConfigMap, ServiceAccount, ClusterRole, ClusterRoleBinding,
+// CustomResourceDefinition and DaemonSet are created
+func CreateNDMYAML(clientset *kubernetes.Clientset, ns string) error {
```

Comment:
@AmitKumarDas You mean like using the single YAML and parsing the resources from it?. 
Setup code is only checking for minikube status, because we apply / delete the YAML mulitple times to check various scenarios